### ISSUE TITLE
Flaky Windows tests

### ISSUE THREAD

**Issue by stefan6419846** [py-pdf/pypdf#2311]
At the moment, the Windows tests seem to be a bit flaky and some tests which rely on https://corpora.tika.apache.org apparently fail randomly. This makes submitting PRs not really smooth.

Some recent examples are https://github.com/py-pdf/pypdf/actions/runs/6996755806/job/19032987483 and https://github.com/py-pdf/pypdf/actions/runs/6987991121/job/19015112409

We should probably try to improve this. Apparently, the retry logic is not sufficient to not get empty files: https://github.com/py-pdf/pypdf/blob/0d0d81b86029fe2313d6d465cc2899dfd4dfb479/tests/__init__.py#L34-L46

**Comment by MartinThoma** [comment]
I agree that the tests seem flaky. I wonder if this is related to the windows workflow. Originally the idea was to download them once (no matter how many different environments we have) and re-use.

I've also introduced `tests/__init__.py::download_test_pdfs` to help with this issue ... or at least make it more obvious what the issue is.

The only real solution is to get test files into `sample-files`. Then we wouldn't have to rely on a third-party web server.

Those specific PDFs cannot be added to `sample-files`, but maybe somebody can craft PDFs that show the described characteristics.

**Comment by stefan6419846** [comment]
On my own fork, I can observe failing Linux tests for the same reasons, so it does not seem to be directly related to Windows. (Actually, the Linux tests on my fork fail far more often than the ones here in the upstream repository, so it does not seem to be limited to the Windows runner for now.)
